File: src/main/java/io/github/biligoldenwater/midiplayer/utils/particlegui/component/Block.java
package io.github.biligoldenwater.midiplayer.utils.particlegui.component;

import io.github.biligoldenwater.midiplayer.utils.particlegui.PixelColor;
import io.github.biligoldenwater.midiplayer.utils.particlegui.interfaces.Component;
import org.bukkit.Color;

public class Block extends Component {
    private PixelColor color;

    public Block(int x, int y, int width, int height) {
        super(x, y, width, height);

        this.color = new PixelColor(Color.BLACK);
    }

    @Override
    public void render() {

    }

    @Override
    public PixelColor getPixel(int x, int y) {
        return color;
    }

    public PixelColor getColor() {
        return color;
    }

    public void setColor(PixelColor color) {
        this.color = color;
    }
}

File: src/main/java/io/github/biligoldenwater/midiplayer/utils/particlegui/component/Block.kt
package io.github.biligoldenwater.midiplayer.utils.particlegui.component

import io.github.biligoldenwater.midiplayer.utils.particlegui.PixelColor
import io.github.biligoldenwater.midiplayer.utils.particlegui.interfaces.Component
import org.bukkit.Color

open class Block(x: Int, y: Int, width: Int, height: Int) : Component(x, y, width, height) {
    var color: PixelColor = PixelColor(Color.BLACK)

    override fun render() {}
    override fun getPixel(x: Int, y: Int): PixelColor? {
        return color
    }
}